### PR TITLE
Added first script for code review test.

### PR DESCRIPTION
**Pull request by choldgraf**
OK, I looked through my code for quite a while and realized that it's really hard to upload a small batch of code because everything seems to be interdependent on everything else. 

I didn't want to upload everything at once, so instead I've opted to load the script which defines my base "data structures" that I use throughout the analysis. I'm now sure if this will be useful or not, largely because it doesn't come with a script that would be easy to run (this might be possible if I added a couple more modules to the repo).

Tell me what you guys think and I can add / remove things that would make this first code review more efficient.

### CODE REVIEW

**Comment by kmanalo** [comment]
Chris,

This looks pretty useful to wade through- I'll do my best to give it a good
lookover in the next couple of days, general impressions etc.

-Kevin

**Comment by kmanalo** [comment]
I'll give it a brief scrub right now assuming you are okay with conforming
to PEP8 convention.

**Comment by choldgraf** [comment]
Of course, please feel free to offer any and all criticisms and
suggestions. This is virgin code, my friends. Never seen by any eyes except
my own inexperienced ones :-)

File: data_structures.py [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+    Sets values of a certain type to a standard number.  This is useful if you want to minimize
+    the impact of extreme values, but don't want to remove the datapoint itself
+    
+    INPUTS:
+        signal          :   (dataframe) the signal that has values you wish to replace
+        vals            :   (index or mask) the value of the index that you wish to replace,
+                            or a mask that identifies indices to replace
+        kind            :   The new value to use in repacement. Must be "median" or an int
+    '''
+    if kind=='median':
```
Comment (kmanalo) [inline data_structures.py]: Trailing whitespace around operator (PEP8)

File: data_structures.py [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+        self.params.hdfloc = savepath + filename + '_data.h5'
+        self.params.paramloc = savepath + filename + '_params.p'
+
+        # Signal-specific
+        self.params.s_rate = srate
+
+        # Initialize signal, metadata, and timebase
+        self.time = nt.UniformTime(length=signal.shape[0], sampling_rate=srate)
+        self.signal = pd.DataFrame(signal, columns=np.arange(signal.shape[1])+1)
+        if time_index==True:
```
Comment (kmanalo) [inline data_structures.py]: Comparison to True should be `if cond is True:` or `if cond:`

File: data_structures.py [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+            modelnotes      :   any extra notes, defaults to 'None'
+        '''
+        if mname in self.modeldict.keys():
+            cont = raw_input('Model already contains this item, overwrite? (y/n): ')
+            if cont == 'y':
+                print('Overwriting...')
+            else:
+                raise NameError('Please rename your model')
+
+        self.modeldict[mname] = {'params':mparams, 'scores':mscores, 'notes':modelnotes}
```
Comment (kmanalo) [inline data_structures.py]: Whitespace after ':'

File: data_structures.py [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+        Shows the results in a model object
+        '''
+        for key, value in self.modeldict.iteritems():
+            if (mname == 'None') | (mname in key):
+                print key
+                
+                for key, value in value.iteritems():
+                    print '----|', key
+                    
+                    if isinstance(value, dict) == True:
```
Comment (kmanalo) [inline data_structures.py]: Comparison to True should be `if cond is True:' or`if cond:`

**Comment by kmanalo** [comment]
This just relates to pure syntax scrubbing.

I ran through a [http://legacy.python.org/dev/peps/pep-0008/](PEP8) parser (using flake8).  I personally use vim-flake8, as a vim user.  See `github.com/nvie/vim-flake8`.  If not a vim user, you can just pip install  the needed tools.  Also, quite useful is [Pep 8 Online](http://pep8online.com/) if you want to copy-paste.

Other nags (less by me but against PEP8 style): line lengths (but it's really your preference),  whitespace in blank lines, 2nd line visual indent (PEP8 checkers are picky here), trailing whitespace, line lengths (PEP8 prefers < 80).

**Comment by kmanalo** [comment]
I need another separate round to give you comments regarding design, and as you said you cover base data structures so there may not be a lot for me to remark here.

I have not really covered standard for docstrings, but this is personal preference if you desire to self-document or conform to a docstring standard:
1. [Numpy Docstrings](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt)
2. [PEP 257](http://legacy.python.org/dev/peps/pep-0257/)

On second review, you already have a pretty consistent standard, and it was helpful in reading some of the functions.

**Comment by kmanalo** [comment]
Again, this is in the territory of your preference, but I noticed that some projects prefer the returning of a string as opposed to printing as a side-effect of the function, as shown in `show_modelresult_models`

Again, I don't know if this is an absolute must, but I guess the best argument is portability/reuse:
[Print a string or return it](http://stackoverflow.com/a/11127439/992834)

File: data_structures.py [hunk 5]
```diff
@@ -0,0 +1,386 @@
...
+    else:
+        try:
+            nval = [kind]*len(signal.columns)
+        except NameError: 
+            print('You need to specify "median", or choose an integer')
+    for i, col in enumerate(signal[vals].columns):
+        signal.loc[vals, col] = nval[i]
+
+def create_df_from_trial_secs(trials, signal_ref='None', s_rate=1000):
+    '''This takes an input of an array with 3 columns: trial begin (s), trial end (s), and trial type.
```
Comment (kmanalo) [inline data_structures.py]: Didn't know if you wanted to try to go the named constant style 

```
TRIAL_BEGIN = 0
TRIAL_END = 1
TRIAL_TYPE = 2
```

Some discussions about constants here: http://stackoverflow.com/questions/2682745/creating-constant-in-python

**Comment by kmanalo** [comment]
Overall, you present a very nice code structure. 
Sorry for my delay in the review, I look forward to seeing how the data structure works in practice.

**Comment by choldgraf** [comment]
Hey Kevin - sorry for being slow to get back on this. The end of the
semester is approaching rapidly, so things are all slamming together at
once :P

I appreciate the comments regarding PEP8 and design. I'll make a pass
through this material with a PEP8 parser, and will see what it spits back
at me. As for general code design, I'll see what I can do about fixing the
truth statements and such.

So perhaps the next step will be to add a bit more code to whats up here
now? I could include some more modules, and ultimately an ipython notebook
that shows the code in action...

Chris

**Comment by kmanalo** [comment]
Hey Chris,

Sorry for the delay, I too have had similar deadlines (May 1st is a big
target)- I'll be more proactive in the next code insert.

Thanks again for keeping up with it; when you can just add the next set of
code where I can interact with it a bit more...

Regards,
Kevin

**Comment by choldgraf** [comment]
Hey Kevin - hope your deadlines came and went well.

Finally finished with semester craziness and I'm going through the script
that I sent you with a PEP8 linter now. I'll get the next batch of code to
you soon. After that, perhaps we can schedule another skype to get back on
track?

Chris

**Comment by kmanalo** [comment]
Things have normalized a bit here- it should be a bit easier this and next
month for me here.

Sure, I'm game for the next set of code.

Let's schedule a skype talk after the next set of scripts.

**Comment by choldgraf** [comment]
Hey - I just committed a new module to the repo. This one is (hopefully) a
bit more general, and expects either arrays or dataframes instead of
defining my own classes. It's a general set of signal processing tools that
I've found useful though many of them are just wrappers for packages that
are already out there).

Let me know if this is clear enough, and I'm happy to chat soon.

Chris

**Comment by kmanalo** [comment]
Hey Chris,

Sorry I haven't replied, I'll be looking at your code this week :)

Regards,
Kevin

**Comment by choldgraf** [comment]
No problem, thanks :)

**Comment by choldgraf** [comment]
Closing this because it's super old, and because I can actually do PRs decently well now thanks to MNE python :)

**Pull request closed without merging**